src/colmap/controllers/base_option_manager.h

```diff
@@ -176,4 +212,26 @@ void BaseOptionManager::RegisterOption(const std::string& name,
   }
 }
 
+template <typename EnumT>
+void BaseOptionManager::AddAndRegisterDefaultEnumOption(
+    const std::string& name,
+    EnumT* option,
+    std::string_view (*to_string_fn)(EnumT),
+    EnumT (*from_string_fn)(std::string_view),
+    const std::string& help_text) {
+  // Create storage for this enum option (unique_ptr for pointer stability)
+  auto info = std::make_unique<EnumOptionInfo>();
+  info->value = std::string(to_string_fn(*option));
+
+  EnumOptionInfo* info_ptr = info.get();
+  info->apply = [info_ptr, option, from_string_fn]() {
+    *option = from_string_fn(info_ptr->value);
+  };
+
+  // Register as a string option pointing to our storage
+  AddDefaultOption(name, &info->value, help_text);
```

Comment:
Wondering whether we can use the existing enum macros to automatically compile a help text with all the available enum values as `"{VALUE_A, VALUE_B, ...}"`?